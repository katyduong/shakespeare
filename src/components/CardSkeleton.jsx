import React from 'react';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Card from 'react-bootstrap/Card';
import Skeleton from 'react-loading-skeleton';

const CardSkeleton = () => {
  const COUNT = 12;
  const repeatingSkeleton = [];
  for (let i = 0; i < COUNT; i += 1) {
    repeatingSkeleton.push(
      <Col key={i}>
        <Card>
          <Card.Header as="h5" className="align-items-center">
            <Skeleton />
          </Card.Header>
          <Card.Body>
            <Card.Title>
              {' '}
              <Skeleton />
            </Card.Title>
            <Card.Text>
              <Skeleton count={2} />
            </Card.Text>
          </Card.Body>
          <Card.Footer>
            <small className="text-muted">
              <Skeleton />
            </small>
          </Card.Footer>
        </Card>
      </Col>
    );
  }
  return (
    <div className="container-fluid">
      <div className="d-flex justify-content-end">
        <Skeleton className="my-4" width={90} />
      </div>
      <Row xs={1} md={2} lg={3} xl={4} className="g-4">
        {repeatingSkeleton}
      </Row>
    </div>
  );
};

export default CardSkeleton;
