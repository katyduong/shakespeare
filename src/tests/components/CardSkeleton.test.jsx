import React from 'react';
import renderer from 'react-test-renderer';
import CardSkeleton from '../../components/CardSkeleton';

test('renders CardSkeleton', () => {
  const tree = renderer.create(<CardSkeleton />).toJSON();
  expect(tree).toMatchSnapshot();
});
