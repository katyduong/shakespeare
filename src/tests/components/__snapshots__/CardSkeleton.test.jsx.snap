// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardSkeleton 1`] = `
Array [
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col"
  >
    <div
      className="card"
    >
      <h5
        className="align-items-center card-header"
      >
        <span>
          <span
            className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
            style={Object {}}
          >
            ‌
          </span>
        </span>
      </h5>
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
           
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </div>
        <p
          className="card-text"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </p>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          <span>
            <span
              className="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
              style={Object {}}
            >
              ‌
            </span>
          </span>
        </small>
      </div>
    </div>
  </div>,
]
`;
